clang/lib/Headers/arm_acle.h

```diff
@@ -752,10 +752,57 @@ __arm_st64bv0(void *__addr, data512_t __value) {
 #define __arm_mte_ptrdiff(__ptra, __ptrb) __builtin_arm_subp(__ptra, __ptrb)
 
 /* Memory Operations Intrinsics */
-#define __arm_mops_memset_tag(__tagged_address, __value, __size)    \
+#define __arm_mops_memset_tag(__tagged_address, __value, __size)               \
   __builtin_arm_mops_memset_tag(__tagged_address, __value, __size)
 #endif
 
+/* Coprocessor Intrinsics */
+#if (!__thumb__ || __thumb2__) && __ARM_ARCH >= 4
```

Comment:
I think that if we are wrapping the version in to the same clause I would rather that we split this condition into:

```suggestion
#if (!__thumb__ || __thumb2__)
#if __ARM_ARCH >= 4
```

That makes all the versions uniform and makes it more obvious that the ARM or Thumb2 mode applies to all versions.